service-spec/service-spec-commons/src/main/java/io/hyscale/servicespec/commons/builder/EffectiveServiceSpecBuilder.java

```diff
@@ -36,75 +38,77 @@
-		return this;
-	}
+    public EffectiveServiceSpecBuilder type(ServiceInputType serviceInputType) {
+        this.serviceInputType = serviceInputType;
+        return this;
+    }
+
+    public EffectiveServiceSpecBuilder withServiceSpec(String serviceSpec) {
+        this.serviceSpec = serviceSpec;
+        return this;
+    }
+
+    public EffectiveServiceSpecBuilder withProfile(String profile) {
+        this.profile = profile;
+        return this;
+    }
 
-	public EffectiveServiceSpecBuilder withProfile(String profile) {
-		this.profile = profile;
-		return this;
-	}
-	
-	public EffectiveServiceSpecBuilder withFieldMetaDataProvider(FieldMetaDataProvider fieldMetaDataProvider) {
+    public EffectiveServiceSpecBuilder withFieldMetaDataProvider(FieldMetaDataProvider fieldMetaDataProvider) {
         this.fieldMetaDataProvider = fieldMetaDataProvider;
         return this;
     }
 
-	public String build() throws HyscaleException {
-
-		validate(serviceSpec, profile);
-
-		if (this.serviceInputType == ServiceInputType.YAML) {
-			// convert to JSON
-			serviceSpec = ServiceSpecBuilderUtil.yamlToJson(serviceSpec);
-			profile = ServiceSpecBuilderUtil.yamlToJson(profile);
-		}
-
-		// Remove unrequired fields from the profile
-		profile = ServiceSpecBuilderUtil.updateProfile(profile);
-
-		String strategicMergeJson = StrategicPatch.apply(serviceSpec, profile, fieldMetaDataProvider);
-
-		return strategicMergeJson;
-	}
-
-	private void validate(String serviceSpec, String profile) throws HyscaleException {
-		if (StringUtils.isBlank(serviceSpec)) {
-			throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_SPEC_REQUIRED);
-		}
-		if (StringUtils.isBlank(profile)) {
-			throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_PROFILE_REQUIRED);
-		}
-		ObjectMapper mapper;
-		if (ServiceInputType.JSON == this.serviceInputType) {
-			mapper = ObjectMapperFactory.jsonMapper();
-		} else {
-			mapper = ObjectMapperFactory.yamlMapper();
-		}
-		try {
-			mapper.readTree(serviceSpec);
-		} catch (IOException e) {
-			throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_SPEC_PARSE_ERROR);
-		}
-		try {
-			mapper.readTree(profile);
-		} catch (IOException e) {
-			throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_PROFILE_PARSE_ERROR);
-		}
-	}
+    public String build() throws HyscaleException {
+
+        validate(serviceSpec, profile);
+
+        if (this.serviceInputType == ServiceInputType.YAML) {
+            // convert to JSON
+            serviceSpec = DataFormatConverter.yamlToJson(serviceSpec);
+            profile = DataFormatConverter.yamlToJson(profile);
+        }
+
+        // Remove unrequired fields from the profile
+        profile = ServiceSpecBuilderUtil.updateProfile(profile);
+
+        String strategicMergeJson = StrategicPatch.apply(serviceSpec, profile, fieldMetaDataProvider);
+
+        return strategicMergeJson;
+    }
+
+    private void validate(String serviceSpec, String profile) throws HyscaleException {
```

Comment:
Add a TODO here to integrate service spec and profile schema validation, as now it only validates the format